ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/chargingProfiles/ChargingProfilesScspServer.kt

```diff
@@ -0,0 +1,103 @@
+    basePathOverride: String? = null
+) : OcpiSelfRegisteringModuleServer(
+    ocpiVersion = VersionNumber.V2_2_1,
+    moduleID = ModuleID.chargingprofiles,
+    interfaceRole = InterfaceRole.SENDER,
+    versionsRepository = versionsRepository,
+    basePathOverride = basePathOverride
+) {
+
+    companion object {
+        const val ACTIVE_CHARGING_PROFILE_CALLBACK_URL = "activechargingprofile/callback"
+        const val CHARGING_PROFILE_CALLBACK_URL = "chargingprofile/callback"
+        const val CLEAR_PROFILE_CALLBACK_URL = "clearprofile/callback"
+        const val PUT_ACTIVE_CHARGING_PROFILE_URL = "activechargingprofile"
+    }
+
+    override suspend fun doRegisterOn(transportServer: TransportServer) {
+        transportServer.handle(
+            method = HttpMethod.POST,
+            path = basePathSegments + listOf(
+                FixedPathSegment(ACTIVE_CHARGING_PROFILE_CALLBACK_URL),
+                VariablePathSegment("requestId")
+            )
+        ) { req ->
+            req.httpResponse {
+                service
+                    .postCallbackActiveChargingProfile(
+                        requestId = req.queryParams["requestId"] ?: "",
+                        result = mapper.readValue(req.body, ActiveChargingProfileResult::class.java)
+                    )
+            }
+        }
+
+        transportServer.handle(
+            method = HttpMethod.POST,
+            path = basePathSegments + listOf(
+                FixedPathSegment(CHARGING_PROFILE_CALLBACK_URL),
+                VariablePathSegment("requestId")
+            )
+        ) { req ->
+            req.httpResponse {
+                service
+                    .postCallbackChargingProfile(
+                        requestId = req.queryParams["requestId"] ?: "",
+                        result = mapper.readValue(req.body, ChargingProfileResult::class.java)
+                    )
+            }
+        }
+
+        transportServer.handle(
+            method = HttpMethod.POST,
+            path = basePathSegments + listOf(
+                FixedPathSegment(CLEAR_PROFILE_CALLBACK_URL),
+                VariablePathSegment("requestId")
+            )
+        ) { req ->
+            req.httpResponse {
+                service
+                    .postCallbackClearProfile(
+                        requestId = req.queryParams["requestId"] ?: "",
+                        result = mapper.readValue(req.body, ClearProfileResult::class.java)
+                    )
+            }
+        }
+
+        transportServer.handle(
+            method = HttpMethod.PUT,
+            path = basePathSegments + listOf(
+                FixedPathSegment(PUT_ACTIVE_CHARGING_PROFILE_URL),
+                VariablePathSegment("sessionId")
+            )
+        ) { req ->
+            req.httpResponse {
+                service
+                    .putActiveChargingProfile(
+                        sessionId = req.queryParams["sessionId"] ?: "",
```

Comment:
```suggestion
                        sessionId = req.pathParams["sessionId"].orEmpty(),
```

Reply:
Fixed in a8e1200f04f193a8b2612567462f1655871a1aca